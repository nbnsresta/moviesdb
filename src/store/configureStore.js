import { configureStore } from "@reduxjs/toolkit";
import createSagaMiddleware from "redux-saga";
import rootReducer from "./slices/rootReducer";
import rootSaga from "./sagas/rootSaga";

const isDevelopment = process.env.NODE_ENV !== "production";

const middlewares = [];
const sagaMiddleware = createSagaMiddleware();
middlewares.push(sagaMiddleware);

if (isDevelopment) {
  const { createLogger } = require("redux-logger");
  middlewares.push(createLogger({ collapsed: true }));
}

const storeConfig = () => {
  const store = configureStore({
    reducer: rootReducer,
    middleware: middlewares,
    devTools: isDevelopment
  });
  sagaMiddleware.run(rootSaga);
  return store;
};

export default storeConfig;
