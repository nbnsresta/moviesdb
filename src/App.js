import React from "react";
import SectionRow from "./components/SectionRow";
import { categories } from "./store/constants";

import logo from "./res/logo.svg";
import css from "./app.module.css";

function App() {
  return (
    <div className={css.app}>
      <header className={css.appHeader}>
        <img src={logo} className={css.appLogo} alt="logo" />
        {/* <SearchBar /> */}
      </header>
      <main>
        <SectionRow title="Trending Movies" category={categories.trending} />
        <SectionRow title="New Movies" category={categories.new} />
      </main>
    </div>
  );
}

export default App;
