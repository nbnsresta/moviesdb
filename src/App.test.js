import React from "react";
import { Provider } from "react-redux";
import { render } from "enzyme";
import App from "./App";
import configureStore from "./store/configureStore";

let store = configureStore();
beforeEach(() => {});
test("renders learn react link", () => {
  const wrapper = render(
    <Provider store={store}>
      <App />
    </Provider>
  );

  expect(wrapper.html()).toMatchSnapshot();
});
