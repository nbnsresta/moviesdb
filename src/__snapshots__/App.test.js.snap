// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `"<header class=\\"appHeader\\"><img src=\\"logo.svg\\" class=\\"appLogo\\" alt=\\"logo\\"></header><main><div class=\\"sectionRow\\"><h2 class=\\"sectionTitle\\">Trending Movies</h2><div class=\\"react-multi-carousel-list  \\"><ul class=\\"react-multi-carousel-track \\" style=\\"transition:none;overflow:unset;transform:translate3d(0px,0,0)\\"></ul></div></div><div class=\\"sectionRow\\"><h2 class=\\"sectionTitle\\">New Movies</h2><div class=\\"react-multi-carousel-list  \\"><ul class=\\"react-multi-carousel-track \\" style=\\"transition:none;overflow:unset;transform:translate3d(0px,0,0)\\"></ul></div></div><div class=\\"sectionRow\\"><h2 class=\\"sectionTitle\\">Upcoming Movies</h2><div class=\\"react-multi-carousel-list  \\"><ul class=\\"react-multi-carousel-track \\" style=\\"transition:none;overflow:unset;transform:translate3d(0px,0,0)\\"></ul></div></div></main>"`;
